inst/rmarkdown/templates/oup/resources/template.tex

```diff
@@ -1,8 +1,31 @@
+% Colors for colorlinks
 $if(colorlinks)$
 \PassOptionsToPackage{dvipsnames,svgnames*,x11names*}{xcolor}
 $endif$
 
-\documentclass[$if(fontsize)$$fontsize$,$else$12pt,$endif$$if(spacing)$$spacing$,$else$halfline,$endif$$if(papersize)$$papersize$,$else$a4paper,$endif$$for(classoptions)$$classoptions$$sep$,$endfor$]{ouparticle}
+% Set document class options
+\documentclass[webpdf$if(papersize)$,$papersize$$else$,large$endif$$if(document_style)$,$document_style$$else$,contemporary$endif$$if(unnumsec)$,unnumsec$endif$$if(namedate)$,namedate$endif$$if(classoptions)$,$for(classoptions)$$classoptions$$sep$,$endfor$$endif$]{oup-authoring-template}
+
+% one column
+$if(onecolumn)$\onecolumn$endif$
+
+%\usepackage{showframe}
+
+% line numbers
+$if(numberlines)$\usepackage{lineno}\linenumbers$endif$
```

Comment:
Good question. I threw this code in the template in case one day we wanted to use it, but never made reference to it in the Rmd template as I was unsatisfied with it and preferred to suggest that as a `header-includes`. If we were to do it, then I think the correct way to go about it so as to be consistent with `number_sections` would be to include it as a `number_lines` variable in the formatting function that is translated in `numberlines` pandoc arg in the function. We would also need a `number_lines_options`:

```
output:
  rticles::oup_article:
    number_lines: TRUE
    number_lines_options: ["mathlines","switch"]
```

Alternatively, we could make the `number_lines` argument do double duty, but that might be confusing:

```
output:
  rticles::oup_article:
    number_lines: ["mathlines","switch"]
```

In either case, I am not sure how to convert a vector of values in the formatting function into variables passed to pandoc (I do this using `rmarkdown::pandoc_variable_arg` for single value variables, but not sure what the appropriate syntax is for multiple value variables.

Reply:
>In either case, I am not sure how to convert a vector of values in the formatting function into variables passed to pandoc (I do this using rmarkdown::pandoc_variable_arg for single value variables, but not sure what the appropriate syntax is for multiple value variables.

I don't think you can pass such values using command line `-V` flag. From doc, https://pandoc.org/MANUAL.html, this would require using a [metadata file ](https://pandoc.org/MANUAL.html#option--metadata-file). So it gets more complicated on the code base part.

And that is also why we are trying to pass most option as base level of YAML header because Pandoc knows how to parse this YAML header and can handle better much complex variable. 

This would be

````
numberlines: TRUE
numberlines-options: ["mathlines","switch"]
````

and the template would work I guess.

Reply:
Can we meet half way and underlying in the template I use numberlines and numberlines-options, but in the format function also create the possibility of using `number_lines` and `number_lines_options: "mathlines,switch"`? I really like keeping format specific stuff in the format function (line numbering is, admittedly, pretty generic, but it wouldn't make sense, for example, in a HTML document...).